libvast/src/file.cpp

```diff
@@ -102,8 +102,10 @@ bool file::read(void* sink, size_t bytes, size_t* got) {
   return is_open_ && detail::read(handle_, sink, bytes, got);
 }
 
-bool file::write(const void* source, size_t bytes, size_t* put) {
-  return is_open_ && detail::write(handle_, source, bytes, put);
+caf::error file::write(const void* source, size_t bytes, size_t* put) {
```

Comment:
```suggestion
[[nodiscard]] caf::error file::write(const void* source, size_t bytes, size_t* put) {
```

Reply:
Also, all `void*` + `size_t` APIs predate the `span<bytes>` era. The right API for this function in my opinion looks like this:

```cpp
[[nodiscard]] caf::expected<size_t> file::write(span<const byte> bytes);
```

However, I think we should rather have a separate refactoring story that goes through all APIs of this kind and do the replacement.

Reply:
I think we should tackle these as we go. A story like that will never be prioritized.